From 7346d86b355871ef80969ccad0876d8a26f278c3 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafaelpalomaravalos@gmail.com>
Date: Sun, 15 Nov 2020 21:30:34 +0100
Subject: [PATCH] Adding battery status

---
 blocks.def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/blocks.def.h b/blocks.def.h
index 9c22d68..ce3ed6b 100644
--- a/blocks.def.h
+++ b/blocks.def.h
@@ -1,8 +1,8 @@
 //Modify this file to change what commands output to your statusbar, and recompile using the make command.
 static const Block blocks[] = {
 	/*Icon*/	/*Command*/		/*Update Interval*/	/*Update Signal*/
+	{"Bat:", "battery.sh", 30, 0},
 	{"Mem:", "free -h | awk '/^Mem/ { print $3\"/\"$2 }' | sed s/i//g",	30,		0},
-
 	{"", "date '+%b %d (%a) %I:%M%p'",					5,		0},
 };
 
-- 
2.26.2

