From 7a15235114cb84af462776e6f908ffdc3e077652 Mon Sep 17 00:00:00 2001
From: Christoph Lohmann <20h@r-36.net>
Date: Tue, 21 Jan 2014 19:43:40 +0100
Subject: [PATCH 01/36] Adding conforming documentation of the commandline
 color support.

---
 tabbed.1 | 16 ++++++++++++----
 tabbed.c | 10 +++++-----
 2 files changed, 17 insertions(+), 9 deletions(-)

diff --git a/tabbed.1 b/tabbed.1
index 0ae29ce..177eacd 100644
--- a/tabbed.1
+++ b/tabbed.1
@@ -14,6 +14,14 @@ tabbed \- generic tabbed interface
 .IR name ]
 .RB [ \-p
 .IR [ s +/- ] pos ]
+.RB [ \-t
+.IR selbgcol ]
+.RB [ \-T
+.IR selfgcol ]
+.RB [ \-u
+.IR normbgcol ]
+.RB [ \-U
+.IR normfgcol ]
 .RB [ \-r
 .IR narg ]
 .IR [ command ... ]
@@ -70,19 +78,19 @@ with the window id, rather than appending it to the end.
 .B \-s
 will disable automatic spawning of the command.
 .TP
-.BI \-t " color"
+.BI \-t " selbgcol"
 defines the selected background color.
 .IR #RGB ,
 .IR #RRGGBB ,
 and X color names are supported.
 .TP
-.BI \-T " color"
+.BI \-T " selfgbcol"
 defines the selected foreground color.
 .TP
-.BI \-u " color"
+.BI \-u " normbgcol"
 defines the normal background color.
 .TP
-.BI \-U " color"
+.BI \-U " normfgcol"
 defines the normal foreground color.
 .TP
 .B \-v
diff --git a/tabbed.c b/tabbed.c
index d30206b..2c388ee 100644
--- a/tabbed.c
+++ b/tabbed.c
@@ -1240,11 +1240,6 @@ main(int argc, char *argv[]) {
 	case 's':
 		doinitspawn = False;
 		break;
-	case 'v':
-		die("tabbed-"VERSION", © 2009-2012"
-			" tabbed engineers, see LICENSE"
-			" for details.\n");
-		break;
 	case 't':
 		selbgcolor = EARGF(usage());
 		break;
@@ -1257,6 +1252,11 @@ main(int argc, char *argv[]) {
 	case 'U':
 		normfgcolor = EARGF(usage());
 		break;
+	case 'v':
+		die("tabbed-"VERSION", © 2009-2012"
+			" tabbed engineers, see LICENSE"
+			" for details.\n");
+		break;
 	default:
 	case 'h':
 		usage();
-- 
2.26.2

