From 0728caee306423cd1dc139e000828329c9285ab7 Mon Sep 17 00:00:00 2001
From: Markus Teich <markus.teich@stusta.mhn.de>
Date: Tue, 10 Mar 2015 11:29:43 +0100
Subject: [PATCH 14/36] fix bug in unmanage: check if lastsel is initialized

Signed-off-by: Christoph Lohmann <20h@r-36.net>
---
 tabbed.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tabbed.c b/tabbed.c
index d24b299..b15f9cb 100644
--- a/tabbed.c
+++ b/tabbed.c
@@ -1125,7 +1125,7 @@ unmanage(int c) {
 			lastsel--;
 		}
 
-		if(c == sel) {
+		if(c == sel && lastsel >= 0) {
 			focus(lastsel);
 		} else {
 			if(sel > c)
-- 
2.26.2

