From 172b4ed8f77e5f08c0d5d754f02f4f6706feea9a Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafaelpalomaravalos@gmail.com>
Date: Sun, 22 Nov 2020 21:39:30 +0100
Subject: [PATCH 5/7] adding fontawesome to font list

---
 config.def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.def.h b/config.def.h
index 7931d66..8cebf10 100644
--- a/config.def.h
+++ b/config.def.h
@@ -5,7 +5,7 @@ static const unsigned int borderpx  = 1;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
-static const char *fonts[]          = { "monospace:size=10" };
+static const char *fonts[]          = { "monospace:size=10", "FontAwesome:size=12" };
 static const char dmenufont[]       = "monospace:size=10";
 static const char col_gray1[]       = "#222222";
 static const char col_gray2[]       = "#444444";
-- 
2.26.2

